src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -475,9 +475,15 @@ private void AppendLineNoPromptInjection(ITextBuffer buffer)
             }
 
             /// <summary>Implements <see cref="IInteractiveWindowOperations2.TypeChar"/>.</summary>
-            internal void TypeChar(char typedChar)
+            public void TypeChar(char typedChar)
             {
-                InsertText(typedChar.ToString());
+                using (var transaction = UndoHistory?.CreateTransaction(InteractiveWindowResources.TypeChar))
```

Comment:
@jasonmalinowski The merging did not work before the change. So I made the change on Delete() and added TypeChar handling in hope to make undo/redo works better (everything was `Text Buffer Change` before the change). I'd love to have this fixed if you can show me the way :)